### ISSUE TITLE
Need feedback when user enters ThingsBoard parameters

### ISSUE THREAD

**Issue by mhlmhl** [glossyio/traffic-monitor#21]
<img width="156" alt="2024-09-17 12_00_38-Node-RED Dashboard — Mozilla Firefox" src="https://github.com/user-attachments/assets/5ed25a6c-8a88-4b6d-8866-6df179e18ee9">
The "Iot Hub (ThingsBoard) / ThingsBoard Parameters" setup screen gives no feedback when the user enters the parameters.  See attached image.

Current experience: user enters 3 fields (URL, port, access token) and hits submit.  All three fields are then blanked out with no indication whether the entered data is correct or not.

- UI should give feedback indicating success (ability to connect to TB with supplied parameters) or failure.
- Ideally, the UI should indicate which field is wrong, e.g. by outlining it in red. This may not be possible in the case of a timeout, but it might be possible to distinguish particular errors, such as DNS failure when looking up the server URL.
- Rather than automatically blanking out all three fields, it would be much kinder to users if they previously-entered parameters are left unchanged so that the user can just correct the fields that are wrong.
- I'm told that the UI automatically defaults the port if left blank. That should be noted on the screen.

**Comment by mhlmhl** [comment]
[42409102224.pdf](https://github.com/user-attachments/files/17033797/42409102224.pdf)

**Comment by glossyio** [comment]
I agree- This would be worth taking a look at again.  Some thoughts:

- This should more tightly be integrated with the `node-red-project/environment` file.  Perhaps we should only handle these variables through the environment file, similar to [Frigate's config](https://docs.frigate.video/configuration/reference)?
- Right now we're only using ThingsBoard HTTP APIs, but we should plan to have variables for MQTT APIs, too.  So, we'll need port designations for both.
- We'd also need a fleshed out UX, pre-filled fields, etc.  
- Agree on test/success/failure indicator for it.

Maybe we handle this as we move towards Node-RED Dashboard-2? Since [node-red-dashboard is deprecated](https://github.com/node-red/node-red-dashboard).

**Comment by glossyio** [comment]
This was substantially refactored with the linked config file PR. Create a new Issue for additional front-end config discussion.
